service/src/main/java/com/ge/predix/acs/encryption/Encryptor.java

```diff
@@ -0,0 +1,9 @@
+package com.ge.predix.acs.encryption;
+
+public interface Encryptor {
```

Comment:
To make this generally usable, we could build this as a credential store interface. It has 2  functions storage and encryption/decryption.  This service could be used as-is with audit or any other credential which ACS needs to store.

`public interface CredentialStore {

    String encryptAndStore(String id, String key, String value);

    String findAndDecrypt(String id, String key, String encrypted);

}`

* id - will be used for lookup and storage
* key -  for encryption/decryption OR authN token in case of vault)